### PR TITLE
[GAIA Compiler] Build algebra layer structures for filter and project in `GraphBuilder`

### PR DESCRIPTION
**Pull request by shirly121**
## Related issue number

Fixes

### CODE REVIEW

File: interactive_engine/compiler/src/test/java/com/alibaba/graphscope/common/ir/FilterTest.java [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+                                builder.literal(10));
+        RelNode filter = builder.filter(equal).build();
+        Assert.assertEquals(
+                "GraphLogicalSource(tableConfig=[{isAll=false, tables=[person]}], alias=[~DEFAULT],"
+                        + " opt=[VERTEX])",
+                filter.explain().trim());
+    }
+
+    /**
+     * 10 > 20 -> always returns false, create {@link LogicalValues} which carries all data types of the node before the filter
```
Comment (longbinlai) [inline interactive_engine/compiler/src/test/java/com/alibaba/graphscope/common/ir/FilterTest.java:109]: Why the behavior?

File: interactive_engine/compiler/src/test/java/com/alibaba/graphscope/common/ir/ProjectTest.java [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+        RexNode variable =
+                builder.source(
+                                new SourceConfig(
+                                        GraphOpt.Source.VERTEX,
+                                        new LabelConfig(false).addLabel("person"),
+                                        "a"))
+                        .variable("a", "age");
+        Project project = (Project) builder.project(ImmutableList.of(variable)).build();
+        Assert.assertEquals("[a.age]", project.getProjects().toString());
+        Assert.assertEquals("RecordType(INTEGER age)", project.getRowType().toString());
```
Comment (longbinlai) [inline interactive_engine/compiler/src/test/java/com/alibaba/graphscope/common/ir/ProjectTest.java:63]: Where are we getting the DataType of age?

File: interactive_engine/compiler/src/test/java/com/alibaba/graphscope/common/ir/ProjectTest.java [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+                                        "a"))
+                        .variable("a");
+        Project project = (Project) builder.project(ImmutableList.of(variable)).build();
+        Assert.assertEquals("[a]", project.getProjects().toString());
+        Assert.assertEquals(
+                "RecordType(Graph_Schema_Type(BIGINT id, CHAR(1) name, INTEGER age) a)",
+                project.getRowType().toString());
+    }
+
+    // project("a.age") -> expr: "a.age", alias: "age"
```
Comment (longbinlai) [inline interactive_engine/compiler/src/test/java/com/alibaba/graphscope/common/ir/ProjectTest.java:50]: Why `alias: age` ?

File: interactive_engine/compiler/src/test/java/com/alibaba/graphscope/common/ir/ProjectTest.java [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+                                        GraphOpt.Source.VERTEX,
+                                        new LabelConfig(false).addLabel("person"),
+                                        "a"))
+                        .variable("a", "age");
+        Project project = (Project) builder.project(ImmutableList.of(variable)).build();
+        Assert.assertEquals("[a.age]", project.getProjects().toString());
+        Assert.assertEquals("RecordType(INTEGER age)", project.getRowType().toString());
+    }
+
+    // project("a.age+1") -> expr: "a.age+1", alias: "$f0"
```
Comment (longbinlai) [inline interactive_engine/compiler/src/test/java/com/alibaba/graphscope/common/ir/ProjectTest.java:66]: What is `$f0` standing for?

File: interactive_engine/compiler/src/main/java/com/alibaba/graphscope/common/ir/tools/config/PathExpandConfig.java [hunk 5]
```diff
@@ -93,6 +97,27 @@ public RelNode getGetV() {
         return getV;
     }
 
+    @Override
+    public String toString() {
+        return "PathExpandConfig{"
```
Comment (longbinlai) [inline interactive_engine/compiler/src/main/java/com/alibaba/graphscope/common/ir/tools/config/PathExpandConfig.java:102]: Always use `StringBuilder` for such long string concatenation

File: interactive_engine/compiler/src/main/java/com/alibaba/graphscope/common/ir/rel/graph/GraphLogicalExpand.java [hunk 6]
```diff
@@ -28,32 +28,26 @@
...
     }
 
     public static GraphLogicalExpand create(
             GraphOptCluster cluster, List<RelHint> hints, RelNode input, TableConfig tableConfig) {
         return new GraphLogicalExpand(cluster, hints, input, tableConfig);
     }
 
-    private GraphOpt.Expand directionOpt() {
+    public GraphOpt.Expand getOpt() {
         ObjectUtils.requireNonEmpty(hints);
```
Comment (longbinlai) [inline interactive_engine/compiler/src/main/java/com/alibaba/graphscope/common/ir/rel/graph/GraphLogicalExpand.java:43]: What is the behavior of `requireNonEmpty`? Throwing error, or quick the program. How does it affect the working thread?

File: interactive_engine/compiler/src/main/java/com/alibaba/graphscope/common/ir/rel/graph/GraphLogicalGetV.java [hunk 7]
```diff
@@ -28,32 +28,26 @@
...
 
     public static GraphLogicalGetV create(
             GraphOptCluster cluster, List<RelHint> hints, RelNode input, TableConfig tableConfig) {
         return new GraphLogicalGetV(cluster, hints, input, tableConfig);
     }
 
-    private GraphOpt.GetV getVOpt() {
+    public GraphOpt.GetV getOpt() {
         ObjectUtils.requireNonEmpty(hints);
         RelHint optHint = hints.get(0);
```
Comment (longbinlai) [inline interactive_engine/compiler/src/main/java/com/alibaba/graphscope/common/ir/rel/graph/GraphLogicalGetV.java:44]: What is `hints`?

File: interactive_engine/compiler/src/test/java/com/alibaba/graphscope/common/ir/ProjectTest.java (hunk 3 above)
Comment (shirly121) [reply interactive_engine/compiler/src/test/java/com/alibaba/graphscope/common/ir/ProjectTest.java:50]: project/aggregate都使用了calcite的推断方式：
1. 如果expression是variable类型，则尽量使用variable中的tag/property来命名，i.e. a->a, a.age -> age ;
2. 对于其他的expression，则通过`$f`前缀加上id来命名，其中id是从0开始直到推断出一个不重复的命名为止

File: interactive_engine/compiler/src/test/java/com/alibaba/graphscope/common/ir/ProjectTest.java (hunk 4 above)
Comment (shirly121) [reply interactive_engine/compiler/src/test/java/com/alibaba/graphscope/common/ir/ProjectTest.java:66]: 同上

File: interactive_engine/compiler/src/main/java/com/alibaba/graphscope/common/ir/rel/graph/GraphLogicalExpand.java (hunk 6 above)
Comment (shirly121) [reply interactive_engine/compiler/src/main/java/com/alibaba/graphscope/common/ir/rel/graph/GraphLogicalExpand.java:43]: 只是抛出异常，单次query返回一个异常信息，不会影响整个服务

File: interactive_engine/compiler/src/test/java/com/alibaba/graphscope/common/ir/ProjectTest.java (hunk 2 above)
Comment (shirly121) [reply interactive_engine/compiler/src/test/java/com/alibaba/graphscope/common/ir/ProjectTest.java:63]: `project.getRowType()`返回所有columns的类型信息，是一个`RelRecord(List<RelDataTypeField>)`类型，每一个expression都一一对应一个`RelDataTypeField`，`RelDataTypeField.getType()`就能拿到对应expression的类型

File: interactive_engine/compiler/src/main/java/com/alibaba/graphscope/common/ir/tools/config/PathExpandConfig.java (hunk 5 above)
Comment (shirly121) [reply interactive_engine/compiler/src/main/java/com/alibaba/graphscope/common/ir/tools/config/PathExpandConfig.java]: done

**Review by longbinlai — APPROVED** [review]
LGTM

**Pull request merged**